src/sentry/snuba/metrics/extraction.py

```diff
@@ -1042,12 +1050,12 @@ def __init__(
         environment: Optional[str] = None,
         groupbys: Optional[Sequence[str]] = None,
         spec_type: MetricSpecType = MetricSpecType.SIMPLE_QUERY,
-        use_updated_env_logic: bool = True,
+        spec_version: Optional[dict[str, bool]] = None,
```

Comment:
Even though it is optional here, we default it to `SPEC_VERSIONS[0]`.

Also, do you think we should name this something like `spec_options` or `spec_configuration`?

Reply:
I would still call it a version since a version can be seen as a variant of the spec which is controlled by some external params.

Regarding the default version, we should default to something similar to `Version(0)` which I specified before.